client/wildcard/src/components/Menu/MenuButton.tsx

```diff
@@ -1,34 +1,52 @@
-import React, { useMemo } from 'react'
+import { ReactNode, useMemo, forwardRef } from 'react'
 
 import { MenuButton as ReachMenuButton } from '@reach/menu-button'
 import { uniqueId } from 'lodash'
 
 import { ForwardReferenceComponent } from '../../types'
 import { Button, ButtonProps } from '../Button'
-import { PopoverTrigger } from '../Popover'
+import { PopoverTrigger, PopoverTriggerProps } from '../Popover'
 
-export type MenuButtonProps = Omit<ButtonProps, 'as'>
+export interface MenuButtonProps extends Omit<ButtonProps, 'children'>, PopoverTriggerProps {}
 
 /**
  * Wraps a styled Wildcard `<Button />` component that can
  * toggle the opening and closing of a dropdown menu.
  *
  * @see — Docs https://reach.tech/menu-button#menubutton
  */
-export const MenuButton = React.forwardRef(({ children, id, ...props }, reference) => {
+export const MenuButton = forwardRef(({ id, children, ...props }, reference) => {
     // To fix rule: "duplicate-id-active"
     // Document has active elements with the same id attribute: menu-button--menu
     const uniqueMenuId = useMemo(() => id ?? uniqueId('menu-button-'), [id])
 
     // We unset aria-controls as it causes accessibility issues if the Menu is not yet rendered in the DOM.
     // aria-controls has low support across screen readers so this shouldn't be an issue: https://github.com/w3c/aria/issues/995
     return (
-        <ReachMenuButton ref={reference} as={PopoverTriggerButton} id={uniqueMenuId} {...props} aria-controls="">
-            {children}
-        </ReachMenuButton>
+        <ReachMenuButton
+            ref={reference}
+            as={PopoverTriggerButton}
+            id={uniqueMenuId}
+            aria-controls=""
+            // Pass empty string in order to suppress TS issue - ReachMenuButton always
+            // should be call with children props
+            // eslint-disable-next-line react/no-children-prop
+            children=""
+            // Pass real children prop with separate special prop to the PopoverTriggerButton because
+            // reach-ui enforces prop-types check and in order to avoid this internal check we are
+            // avoiding passing real children with standard children prop
+            childrenContent={children}
```

Comment:
Is my understanding correct that we introduce the extra `childrenContent` prop to bypass the Typescript types error? There's no need for it from the runtime perspective – the `children` prop [will be propagated correctly](https://github.com/reach/reach-ui/blob/26c826684729e51e45eef29aa4316df19c0e2c03/packages/menu-button/src/menu-button.tsx#L125) to the `PopoverTrigger` if we use `children` as usual. 

If the above is true, would keeping only a `children` prop along with type casting save the day here? It would be nice to minimize the number of props passed down and the number of layers they go through to simplify the mental model. 

```suggestion
            // `children` are propagated to `PopoverTrigger`, so it's safe to typecast here.
            // eslint-disable-next-line react/no-children-prop
            children={children as ReachMenuButtonProps['children']}
```

Reply:
@valerybugakov 
> we introduce the extra childrenContent prop to bypass the Typescript types error
This isn't quite correct here. The problem there is Typescript and PropTypes internal check that is placed inside of Reach Menu components. According to this check we can't pass anything beyond ReactNode as a children to the ReachMenuButton. But we want to pass render props as a child there. If we just suppress TS problem we'll have a Runtime PropTypes error in a console in a browser.

Reply:
> there is Typescript and PropTypes internal check that is placed inside of Reach Menu components

Oh, so they have a runtime check also? 😬 I thought this issue was limited to Typescript types check that can be bypassed by typecasting. If it's not the case, using an extra prop makes sense.

Reply:
Yeah. ReachUI uses prop-types for props checking, I believe they should be removed from the production bundle but still it produces console.error with invalid prop types in dev bundle which may confuse some ends who are working with this components.